src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -398,9 +401,13 @@ private async void RunLeaseRenewalLoopAsync()
 
         private async Task RenewLeasesAsync(SqlConnection connection, CancellationToken token)
         {
-            try
+            this._logger.LogDebugWithThreadId("BEGIN WaitRowsLock - RenewLeases");
+            await this._rowsLock.WaitAsync(token);
+            this._logger.LogDebugWithThreadId("END WaitRowsLock - RenewLeases");
+
+            if (this._state == State.ProcessingChanges)
```

Comment:
Have single a `if` statement instead of repetitive checks in try and finally blocks.